pandas/core/arrays/arrow/array.py

```diff
@@ -1868,13 +1871,29 @@ def _str_join(self, sep: str):
         return type(self)(pc.binary_join(self._pa_array, sep))
 
     def _str_partition(self, sep: str, expand: bool):
-        raise NotImplementedError(
-            "str.partition not supported with pd.ArrowDtype(pa.string())."
+        return type(self)(
+            pa.chunked_array(
+                [
+                    [
+                        None if val.as_py() is None else val.as_py().partition(sep)
+                        for val in chunk
+                    ]
+                    for chunk in self._pa_array.iterchunks()
```

Comment:
it looks like this will maintain the chunking structure.  is there a reason not to chain these together and end up with a single chunk?